core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -193,102 +193,12 @@ public Result setOpToSql(SqlSetOperator operator, RelNode rel) {
    * @param node            Join condition
    * @param leftContext     Left context
    * @param rightContext    Right context
-   * @param leftFieldCount  Number of fields on left result
    * @return SqlNode that represents the condition
    */
-  public static SqlNode convertConditionToSqlNode(RexNode node,
-      Context leftContext,
-      Context rightContext,
-      int leftFieldCount,
-      SqlDialect dialect) {
-    if (node.isAlwaysTrue()) {
-      return SqlLiteral.createBoolean(true, POS);
-    }
-    if (node.isAlwaysFalse()) {
-      return SqlLiteral.createBoolean(false, POS);
-    }
-    if (node instanceof RexInputRef) {
-      Context joinContext = leftContext.implementor().joinContext(leftContext, rightContext);
-      return joinContext.toSql(null, node);
-    }
-    if (!(node instanceof RexCall)) {
-      throw new AssertionError(node);
```

Comment:
Will Calcite lose the ability to does this assertion?

Reply:
I cannot figure out a counter-example when converting filter condition.
Correct me if I'm wrong.

Reply:
You mean cannot construct a valid SQL string in which a join condition is not a RexCall?

I was thinking I could use RelBuilder to construct join, and then calls a RelToSqlConvert.

Reply:
To make myself clear -- I cannot figure out a join condition, which should not be supported here.

Reply:
I see. I was thinking if this is used to validate "invalidate" input. 

E.g. 
```
SELECT * FROM T JOIN Q ON "RexLiteral"
```

Do you think if such query could be passed in?

Reply:
Well I think this is a valid sql -- Literal of "RexLiteral" turns out to be a `alwaysTrue` condition.

Reply:
I tried a INNER JOIN query with `ON 1` and Calcite validator throws an exception "ON clause must be a condition".

Regrading to the specific assertion we have discussed, it might just minor so either keep it or remove it is fine.